# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [2]:
# TODO
import numpy as np
import scipy.stats as stats

diet_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

t_value, p_value = stats.ttest_ind(diet_1, diet_2)

print(f'Valor de T: {t_value}')
print(f'Valor de P:{p_value:.4f}')

      


Valor de T: -3.5383407969933938
Valor de P:0.0041


CONCLUSIÓN: Siendo la hipótesis nula que no hay diferencia entre las dos dietas se concluye debido a que el valor de p es menor de 0,05 que se rechaza dicha hipótesis y se acepta la hipótesis alternativa, es decir, que las dos dietas tienen efectos diferentes. 

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [4]:
# TODO
Fertilizante_1 = [20, 21, 20, 19, 20]
Fertilizante_2 = [22, 21, 23, 22, 21]
Fertilizante_3 = [24, 23, 22, 23, 24]

f_statistic, p_value = stats.f_oneway(Fertilizante_1, Fertilizante_2, Fertilizante_3)

print(f'Estadístico F: {f_statistic}')
print(f'Valor de P: {p_value}')

Estadístico F: 20.315789473684188
Valor de P: 0.00014047824793190472


CONCLUSIÓN: Al ser el valor de P menos que 0,05 aceptamos la hipótesis alternativa que dice que si que hay un fertilizante que sería superior entérminos de producción del maíz. A su vez, el número F al ser tan alto nos dice que hay una diferencia notable entre los tipos de fertilizantes.

In [7]:
media_fer_1 = np.mean(Fertilizante_1)
media_fer_2 = np.mean(Fertilizante_2)
media_fer_3 = np.mean(Fertilizante_3)

print(f'Media del fertilizante 1: {media_fer_1}')
print(f'Media del fertilizante 2: {media_fer_2}')
print(f'Media del fertilizante 3: {media_fer_3}')

resultados = stats.tukey_hsd(Fertilizante_1, Fertilizante_2, Fertilizante_3)

print(resultados)

Media del fertilizante 1: 20.0
Media del fertilizante 2: 21.8
Media del fertilizante 3: 23.2
Tukey's HSD Pairwise Group Comparisons (95.0% Confidence Interval)
Comparison  Statistic  p-value  Lower CI  Upper CI
 (0 - 1)     -1.800     0.010    -3.143    -0.457
 (0 - 2)     -3.200     0.000    -4.543    -1.857
 (1 - 0)      1.800     0.010     0.457     3.143
 (1 - 2)     -1.400     0.041    -2.743    -0.057
 (2 - 0)      3.200     0.000     1.857     4.543
 (2 - 1)      1.400     0.041     0.057     2.743



CONCLUSIÓN: El mejor fertilizante sería el tercero ya que tiene la media mas alta de los 3 y cuando los comparamos este es mejor que el primero con una diferencia de 3.2 puntos y mejor que el segundo con una diferencia de 1.4 puntos. Le seguiría el fertilizante 2, que es mejor que el primero con una diferencia de 1.8 puntos y tiene la segunda media mas alta.